### PR TITLE
Change date format according to the GLOBAL variables

### PR DESCRIPTION
**Pull request by eyalvo-matrix**
Hi all, 

This fix intended to change the date in the office comments
according to the Date Display Format Global variable. 

Eyal

### CODE REVIEW

File: interface/main/onotes/office_comments.php [hunk 1]
```diff
@@ -54,12 +54,27 @@
...
+        default:
+            $datefornat= "D F dS";
+    }
+
     if (getdate() == strtotime($iter{"date"})) {
-        $date_string = "Today, " . date( "D F dS" ,strtotime($iter{"date"}));
+        $date_string = "Today, " . date( $datefornat ,strtotime($iter{"date"}));
     } else {
-        $date_string = date( "D F dS" ,strtotime($iter{"date"}));
+        $date_string = date( $datefornat ,strtotime($iter{"date"}));
```
Comment (bradymiller) [inline interface/main/onotes/office_comments.php]: Use this function instead here:
https://github.com/openemr/openemr/blob/master/library/formatting.inc.php#L20-L30

File: interface/main/onotes/office_comments_full.php [hunk 2]
```diff
@@ -108,11 +108,27 @@
...
+        default:
+            $datefornat= "D F dS";
+    }
+
     if (getdate() == strtotime($iter{"date"})) {
-        $date_string = "Today, " . date( "D F dS" ,strtotime($iter{"date"}));
+        $date_string = "Today, " . date( $datefornat ,strtotime($iter{"date"}));
     } else {
-        $date_string = date( "D F dS" ,strtotime($iter{"date"}));
+        $date_string = date($datefornat ,strtotime($iter{"date"}));
```
Comment (bradymiller) [inline interface/main/onotes/office_comments_full.php]: Use this function instead here:
https://github.com/openemr/openemr/blob/master/library/formatting.inc.php#L20-L30

File: interface/main/onotes/office_comments_full.php [hunk 3]
```diff
@@ -108,11 +108,27 @@
...
+        case 2:
+            $datefornat= "d/m/y";
+            break;
+        default:
+            $datefornat= "D F dS";
+    }
+
     if (getdate() == strtotime($iter{"date"})) {
-        $date_string = "Today, " . date( "D F dS" ,strtotime($iter{"date"}));
+        $date_string = "Today, " . date( $datefornat ,strtotime($iter{"date"}));
```
Comment (bradymiller) [inline interface/main/onotes/office_comments_full.php]: Use this function instead here:
https://github.com/openemr/openemr/blob/master/library/formatting.inc.php#L20-L30
(no need to support D F dS stuff and rec. translating the Today label)

**Comment by bradymiller** [comment]
Placed a review. Looking forward to the next revision.

File: interface/main/onotes/office_comments.php [hunk 4]
```diff
@@ -54,12 +54,27 @@
...
+        case 2:
+            $datefornat= "d/m/y";
+            break;
+        default:
+            $datefornat= "D F dS";
+    }
+
     if (getdate() == strtotime($iter{"date"})) {
-        $date_string = "Today, " . date( "D F dS" ,strtotime($iter{"date"}));
+        $date_string = "Today, " . date( $datefornat ,strtotime($iter{"date"}));
```
Comment (eyalvo-matrix) [inline interface/main/onotes/office_comments.php]: Hi Brady, Thank you for your advice 
I just committed changes hopefully as you instructed 
Is it good now?

File: interface/main/onotes/office_comments.php (hunk 1 above)
Comment (eyalvo-matrix) [inline interface/main/onotes/office_comments.php]: Hi Brady, Thank you for your advice 
I just committed changes hopefully as you instructed 
Is it good now?

File: interface/main/onotes/office_comments_full.php (hunk 3 above)
Comment (eyalvo-matrix) [inline interface/main/onotes/office_comments_full.php]: Hi Brady, Thank you for your advice 
I just committed changes hopefully as you instructed 
Is it good now?

File: interface/main/onotes/office_comments_full.php (hunk 2 above)
Comment (eyalvo-matrix) [inline interface/main/onotes/office_comments_full.php]: Hi Brady, Thank you for your advice 
I just committed changes hopefully as you instructed 
Is it good now?

**Comment by bradymiller** [comment]
Going to bring this fix into the codebase with a couple change, which will document on the review. So, don't need to fix it, just for educational purposes.

File: interface/main/onotes/office_comments.php [hunk 5]
```diff
@@ -54,15 +55,17 @@
...
     }
-    
-    
+
+    $date=date( "Y-m-d" ,strtotime($iter{"date"}));
+    $date=oeFormatShortDate($date);
+
     if (getdate() == strtotime($iter{"date"})) {
-        $date_string = "Today, " . date( "D F dS" ,strtotime($iter{"date"}));
+        $date_string = xlt("Today, ") . $date;
```
Comment (bradymiller) [inline interface/main/onotes/office_comments.php]: 1. Not html escaping here, so used xl()
2. Constants should not include trailing or leading spacing/punctuation except when end of sentence.
``````
$date_string = xl("Today") . ", " . $date;

File: interface/main/onotes/office_comments_full.php [hunk 6]
```diff
@@ -108,11 +109,14 @@
...
 foreach ($result as $iter) {
     $result_count++;
-    
+
+    $date=date( "Y-m-d" ,strtotime($iter{"date"}));
+    $date=oeFormatShortDate($date);
+
     if (getdate() == strtotime($iter{"date"})) {
-        $date_string = "Today, " . date( "D F dS" ,strtotime($iter{"date"}));
+        $date_string = xlt("Today, ") . $date;
```
Comment (bradymiller) [inline interface/main/onotes/office_comments_full.php]: 1. Not html escaping here, so used xl()
2. Constants should not include trailing or leading spacing/punctuation except when end of sentence.
``````
$date_string = xl("Today") . ", " . $date;

File: interface/main/onotes/office_comments.php [hunk 7]
```diff
@@ -54,15 +55,17 @@
...
     } else {
-        $date_string = date( "D F dS" ,strtotime($iter{"date"}));
+        $date_string = $date;
     }
     
-    print "<tr><td width=20% valign=top><font class='bold'>".text($date_string)."</font> <font class='bold'>(".text($iter{"user"}).")</font><br>" . "<font class='text'>" . text($iter{"body"}) . "</font></td></tr>\n";
+    print "<tr><td width=20% valign=top><font class='bold'>".text($date_string)."</font> <font class='bold'>(".xlt(text($iter{"user"})).")</font><br>" . "<font class='text'>" . text($iter{"body"}) . "</font></td></tr>\n";
```
Comment (bradymiller) [inline interface/main/onotes/office_comments.php]: Removed the translation of user name.

**Comment by bradymiller** [comment]
Brought this into the codebase(with above mods). thanks for the fix!
-brady

File: interface/main/onotes/office_comments.php (hunk 7 above)
Comment (eyalvo-matrix) [inline interface/main/onotes/office_comments.php]: ok

File: interface/main/onotes/office_comments.php (hunk 5 above)
Comment (eyalvo-matrix) [inline interface/main/onotes/office_comments.php]: Thanks I will change that

File: interface/main/onotes/office_comments_full.php (hunk 6 above)
Comment (eyalvo-matrix) [inline interface/main/onotes/office_comments_full.php]: Thanks I will change that

File: interface/main/onotes/office_comments.php (hunk 5 above)
Comment (bradymiller) [inline interface/main/onotes/office_comments.php]: no need to change this. I already did it and brought your fix into the codebase :)
(above is just education)

File: interface/main/onotes/office_comments_full.php (hunk 6 above)
Comment (bradymiller) [inline interface/main/onotes/office_comments_full.php]: no need to change this. I already did it and brought your fix into the codebase :)
(above is just education)

File: interface/main/onotes/office_comments_full.php [hunk 8]
```diff
@@ -121,7 +125,7 @@
     print "<tr><td><input type=hidden value='' name='act".attr($iter{"id"})."' id='act".attr($iter{"id"})."'>";
     print "<input name='box".attr($iter{"id"})."' id='box".attr($iter{"id"})."' onClick='javascript:document.update_activity.act".attr($iter{"id"}).".value=this.checked' type=checkbox $checked></td>";
     print "<td><label for='box".attr($iter{"id"})."' class='bold'>".text($date_string) . "</label>";
-    print " <label for='box".attr($iter{"id"})."' class='bold'>(". text($iter{"user"}).")</label></td>";
+    print " <label for='box".attr($iter{"id"})."' class='bold'>(". xlt(text($iter{"user"})).")</label></td>";
```
Comment (eyalvo-matrix) [inline interface/main/onotes/office_comments_full.php]: ok

**Pull request closed without merging**